dsync/http.go

```diff
@@ -5,21 +5,43 @@ import (
 	"context"
 	"encoding/json"
 	"fmt"
+	"io"
 	"io/ioutil"
 	"net/http"
 	"net/url"
 
+	format "github.com/ipfs/go-ipld-format"
+	coreiface "github.com/ipfs/interface-go-ipfs-core"
+	protocol "github.com/libp2p/go-libp2p-core/protocol"
 	"github.com/qri-io/dag"
 )
 
+const (
+	httpProtcolIDHeader = "dsync-version"
+	sidHeader           = "sid"
+)
+
+const (
+	carMIMEType    = "archive/car"
+	cborMIMEType   = "application/cbor"
+	jsonMIMEType   = "application/json"
+	binaryMIMEType = "application/octet-stream"
+)
```

Comment:
I guess this could live elsewhere or actually be used with https://golang.org/src/mime/type.go
But perfectly fine if this lives on.

Reply:
yeah this is more for spec writing purposes, I wanted to catalog the set of MIME types the protocol uses. I'd like remove the need for JSON entirely